exporters/otlp/src/otlp_environment.cc

```diff
@@ -80,6 +80,34 @@ static bool GetStringDualEnvVar(const char *signal_name,
   return exists;
 }
 
+static std::uint32_t GetUintEnvVarOrDefault(opentelemetry::nostd::string_view signal_env,
+                                            opentelemetry::nostd::string_view generic_env,
+                                            std::uint32_t default_value)
+{
+  std::string value;
+
+  if (GetStringDualEnvVar(signal_env.data(), generic_env.data(), value))
+  {
+    return static_cast<std::uint32_t>(std::strtoul(value.c_str(), nullptr, 10));
```

Comment:
Here, any garbage in the environment variable that does not fit into a ul string will be silently ignored.

Please implement `GetUIntEnvironmentVariable` in sdk/common instead, and log warnings when invalid strings are found.

See existing code for Bool and Duration.